File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int>cmap;
        for(int i= 0;i<nums.size();i++){
            int value = target - nums[i];
            if(cmap.find(value) != cmap.end()){
                return {cmap[value], i};
            } else {
                cmap[nums[i]] = i;
            }
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        values = {}
        res = []
        for ind, num in enumerate(nums):
            k = target - num
            if k in values:
                res.append(values[k])
                res.append(ind)
            values[num] = ind
        
        return res
                